statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,726 @@
+"""
+Created on Wed Jul 12 09:35:35 2017
+
+Notes
+-----
+Code written using below textbook as a reference.
+Results are checked against the expected outcomes in the text book.
+
+Properties:
+Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles and practice. OTexts, 2014.
+
+Author: Terence L van Zyl
+
+"""
+import numpy as np
+
+from statsmodels.base.model import Results
+from statsmodels.base.wrapper import populate_wrapper, union_dicts, ResultsWrapper
+from statsmodels.tsa.base.tsa_model import TimeSeriesModel
+
+from scipy.optimize import basinhopping, brute, minimize
+from scipy.spatial.distance import sqeuclidean
+try:
+    from scipy.special import inv_boxcox
+except ImportError:
+    def inv_boxcox(x, lmbda):
+        return np.exp(np.log1p(lmbda * x) / lmbda) if lmbda != 0 else np.exp(x)
+from scipy.stats import boxcox
+
+
+def _holt_init(x, xi, p, y, l, b):
+    p[xi] = x
+    alpha, beta, _, l0, b0, phi = p[:6];
+    alphac = 1 - alpha
+    betac = 1 - beta
+    y_alpha = alpha * y
+    l[:] = 0; b[:] = 0;
+    l[0] = l0
+    b[0] = b0
+    return alpha, beta, phi, alphac, betac, y_alpha
+
+
+def _holt__(x, xi, p, y, l, b, s, m, n, max_seen):
```

Comment:
why two trailing underscores?

Reply:
It was to help organise it when I was coding.
The spaces between underscores helped with knowing which components of the exponential smoothing model are left out.

Reply:
I think I see what you mean, but (as are as I can tell) the number of underscores isn't consistently used across the functions, so maybe it's best just to eliminate underscores at the end and eliminate double underscores?